### PR TITLE
Support for configurations remapping in solution files

### PR DESCRIPTION
**Pull request by deitry**
Solves #1828

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=1835) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=1835) before we can accept your contribution.<br/><br/>:x: deitry [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-roslyn?pullRequest=1835)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-roslyn?pullRequest=1835) it.</sub>

File: src/OmniSharp.MSBuild/ProjectLoader.cs [hunk 1]
```diff
@@ -115,10 +116,37 @@ public MSB.Evaluation.Project EvaluateProjectFile(string filePath)
...
-        private MSB.Evaluation.Project EvaluateProjectFileCore(string filePath)
+        private MSB.Evaluation.Project EvaluateProjectFileCore(string filePath, Dictionary<string, string> projectConfigurationsInSolution = null)
         {
+            var localProperties = new Dictionary<string, string>(_globalProperties);
+            if (projectConfigurationsInSolution != null
+                && localProperties.TryGetValue(PropertyNames.Configuration, out string solutionConfiguration))
+            {
+                if (!localProperties.TryGetValue(PropertyNames.Platform, out string solutionPlatform))
+                {
+                    solutionPlatform = "Any CPU";
```
Comment (david-driscoll) [inline src/OmniSharp.MSBuild/ProjectLoader.cs:127]: I wonder if there are any other alternate values for Any CPU we should be aware of?

File: src/OmniSharp.MSBuild/ProjectIdInfo.cs [hunk 2]
```diff
@@ -13,5 +14,6 @@ public ProjectIdInfo(ProjectId id, bool isDefinedInSolution)
 
         public ProjectId Id { get; set; }
         public bool IsDefinedInSolution { get; set; }
+        public Dictionary<string, string> ConfigurationsInSolution { get; set; }
```
Comment (mholo65) [inline src/OmniSharp.MSBuild/ProjectIdInfo.cs:17]: Would `SolutionConfiguration` be a better name here?
Consider using `IDictionary` or `IReadOnlyDictionary` instead.

File: src/OmniSharp.MSBuild/ProjectLoader.cs [hunk 3]
```diff
@@ -67,11 +67,12 @@ public ProjectLoader(MSBuildOptions options, string solutionDirectory, Immutable
             return globalProperties;
         }
 
-        public (MSB.Execution.ProjectInstance projectInstance, ImmutableArray<MSBuildDiagnostic> diagnostics) BuildProject(string filePath)
+        public (MSB.Execution.ProjectInstance projectInstance, ImmutableArray<MSBuildDiagnostic> diagnostics) BuildProject(
+            string filePath, Dictionary<string, string> configurationsInSolution)
         {
```
Comment (mholo65) [inline src/OmniSharp.MSBuild/ProjectLoader.cs:72]: Use `IReadOnlyDictionary<string, string>` or `IEnumerable<KeyValuePair<string, string>>`. Or maybe `ImmutableDictionary<string, string>` to make it consistent with current API.

File: src/OmniSharp.MSBuild/ProjectIdInfo.cs (hunk 2 above)
Comment (deitry) [reply src/OmniSharp.MSBuild/ProjectIdInfo.cs:17]: I tried to state that this is not 'configurations of solution' but 'configurations of project in solution'. But you are right, `SolutionConfiguration` is indeed a better name.

**Comment by david-driscoll** [comment]
@JoeRobich @filipw thoughts?

**Pull request merged**